src/cmdstan/command.hpp

```diff
@@ -255,42 +306,30 @@ int command(int argc, const char *argv[]) {
     int num_draws = get_arg_val<int_argument>(*pathfinder_arg, "num_draws");
     int num_psis_draws
         = get_arg_val<int_argument>(*pathfinder_arg, "num_psis_draws");
-    int num_paths = get_arg_val<int_argument>(*pathfinder_arg, "num_paths");
-    bool save_iterations
-        = get_arg_val<bool_argument>(*pathfinder_arg, "save_single_paths");
-
-    if (num_paths == 1) {
-      if (!get_arg_val<string_argument>(parser, "output", "diagnostic_file")
-               .empty()) {
-        save_iterations = true;
-      }
+    if (num_chains == 1) {
+      save_single_paths = save_single_paths || !diagnostic_file.empty();
```

Comment:
I don't understand this change

Are the diagnostic files ever nonempty if there's no saved single paths? I don't fully recall what we decided for the interface, might be worth commenting here.

Reply:
the logic is super-opaque - I don't quite remember either.   I think our reasoning was:

- save_single_paths always gives you the diagnostic file
-if somone specifies a diagnostic file and num_paths == 1,  give them a diagnostic file which has whatever basename they specify.
- if someone specifies num_paths==1, but doesn't want a diagnostic file, don't output one.